core/vulkan/tools/image.cpp

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "image.h"
+
+#define STB_IMAGE_WRITE_IMPLEMENTATION
+#include "stb_image_write.h"
+
+namespace {
+
+// A function of type 'stbi_write_func' that writes |size| bytes of |data| to a
+// stream passed via |context| as a std::ostream pointer.
+void WriteDataToStream(void* context, void* data, int size) {
+  std::ostream* stream = static_cast<std::ostream*>(context);
+  stream->write(static_cast<char*>(data), size);
+}
+
+}  // namespace
+
+namespace vk_tools {
+
+void WritePng(std::ostream* stream, uint8_t* image_data, size_t size,
+              uint32_t width, uint32_t height, VkFormat image_format) {
+  switch (image_format) {
+    case VK_FORMAT_B8G8R8A8_UNORM:
+    case VK_FORMAT_B8G8R8A8_UINT: {
+      // Convert the image data from BGRA to RGBA
+      auto data = image_data;
```

Comment:
As this is now a library function, we should at least check that the width*height*data_size is what we actually expect, and return if the data is wrong. Rather than writing to arbitrary memory.

Reply:
added check